guides-markdown/connect-local-nodes.md

```diff
@@ -0,0 +1,141 @@
+# Connect two local nodes
+
+```post-author
+Orfeas Stefanos Thyfronitis Litos
+```
+
+```post-description
+Start and manually connect two regtest nodes in a single script.
+```
+
+When we start two nodes in the regtest (loopback network, 127.0.0.1) with
+manually set ports, they don't get connected automatically because they don't
+know each other's port. We thus have to connect them manually. In this example
+we will fire up one SPV and one Full node and initiate a connection from the
+SPV side. The SPV node is given the address of the Full node and attempts to
+connect. If everything goes well, the two nodes connect. The Full node then
+broadcasts two transactions. The first tx pays to an address contained in the
+bloom filter of the SPV node, the second tx to another address which the SPV
+node doesn't watch. If everything goes well, the first transaction is received
+by the SPV node and the second isn't. Finally the two nodes are gracefully
+closed.
+
+We use the library `p-event` which promisifies events in order to `await` for
+connection of the nodes and the reception of the transaction cleanly.
+
+Requirements: `bcoin` (duh), `p-event` (`npm install bcoin p-event`)
+
+```javascript
+const bcoin = require('bcoin').set('regtest');
+const NetAddress = bcoin.net.NetAddress;
+const Network = bcoin.Network;
+const pEvent = require('p-event'); // tool to await for events
+
+async function delay(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
+}
+
+const regtest = Network.get().toString();
+
+// create nodes
+const spvNode = new bcoin.SPVNode({
+  network: regtest,
+  httpPort: 48449, // avoid clash of ports
+
+  // write log file and chain data to specific directory
+  prefix: '~/connect-test/SPV',
```

Comment:
small nit, but it might be nice to make `~/connect-test` a constant since it's used in multiple places. This is nice in case someone forgets to change both and ends up with two different project directories. Also if you use node's `os.homedir()` method instead of `~` it can work across platforms since I don't think Windows recognizes `~`.

Reply:
Doesn't windows use backslashes instead of slashes as well? I don't have a windows box to test upon, but I think something in the lines of https://gist.github.com/domenic/2790533 should be used. More info: https://nodejs.org/docs/latest/api/path.html. I think I got it correctly; At least my box is happy. If anyone can test it on a windows box, please tell me if it works for them.

Reply:
Well, the beauty if using Node's `os` module is that it will take care of environment specific conditions for you so you don't even have to worry about it.

Reply:
i.e. `const prefix = `${require('os').homedir()}/connect-test/SPV`
if you look at the bcfg library which manages this in bcoin, it makes heavy use of this convention, even explicitly replacing instances of `~` passed by the user.

Reply:
Thanks for pointing me there, I found `path.join()` which accomplishes what I wanted more simply than `path.format()`. Let me know if you'd like any more changes.